modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/index/contributor/ObjectEntryFolderModelDocumentContributor.java

```diff
@@ -31,6 +36,42 @@ public void contribute(
 		document.addText(Field.NAME, objectEntryFolder.getName());
 		document.addLocalizedKeyword(
 			"localized_label", objectEntryFolder.getLabelMap(), true, true);
+
+		String[] parts = StringUtil.split(
+			objectEntryFolder.getTreePath(), CharPool.SLASH);
+
+		document.addKeyword(Field.TREE_PATH, parts);
```

Comment:
@adolfopa, it might be that Brian will want this after document.addText(Field.NAME, objectEntryFolder.getName()); being addLocalizedKeyword last